src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,98 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const [newMovie, setNewMovie] = useState<Movie>({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const {
+    title, description, imgUrl, imdbUrl, imdbId,
+  } = newMovie;
+
+  const disabled = !(
+    title.trim() && imgUrl.trim() && imdbUrl.trim() && imdbId.trim()
+  );
+
+  const clear = () => {
+    setNewMovie({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+  };
+
+  const handleInputChange
+    = (event: React.ChangeEvent<HTMLInputElement>, field: string) => {
+      setNewMovie({ ...newMovie, [field]: event.target.value });
```

Comment:
```suggestion
      setNewMovie({ ...newMovie, [event.target.name]: event.target.value });
```

you could do it in this way - get input name from the event, instead of `field` argument - FYI